### PR TITLE
Feature/9 park details events UI

### PR DESCRIPTION
**Pull request by SaturneV**
### Summary

This pull request introduces the UI for park details and events, enhancing the user experience by providing detailed information about parks and their events.

<p align="center">
Screenshot 2024-10-09 at 09 19 51
Screenshot 2024-10-09 at 11 19 17

</p>

### Changes

- Implemented UI for displaying park details. (`ParkOverview.kt`)
- Added components for listing upcoming events at each park. (`ParkOverview.kt`)
- Implemented utility function for Timestamp formatting. (`DateUtils.kt`)
- Implemented _draft_ data classes for parks and events. (`Park.kt` and `Event.kt`)

_The data classes are a first draft for the parks and events. Their primary aim is to enable the development of a functional UI component. These classes will need to be adapted to more suitable data classes when the complete MVVM architecture is implemented._

### Testing

- Implemented UI tests for the `ParkOverviewScreen` composable. (`ParkOverviewTest.kt`)

_Currently, the tests only verify if all the elements are displayed. When the park overview UI is implemented in the project, it will be necessary to test its integration, such as handling navigation and actions like button clicks._

### CODE REVIEW

**Review by alvaro1080 — CHANGES_REQUESTED** [review]
Thank you for this PR. The ParkOverview UI is beautiful and will be very useful to create a good user experience. I have a few suggestions and a few typos I would like to highlight:

File: app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+ *
+ * @param image The image to display.
+ * @param title The title to display.
+ */
+@Composable
+fun ImageTitle(image: Painter?, title: String) {
+  Box(modifier = Modifier.fillMaxWidth().height(220.dp).testTag("imageTitle")) {
+    Image(
+        // If no image is provided, use the default park image.
+        painter = image ?: painterResource(id = R.drawable.park_default),
```
[the remark this answers is not in the document]
Comment (SaturneV) [reply app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt:88]: Good point indeed, it's done!

File: app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  Column(modifier = Modifier.testTag("eventItemList")) {
+    Text(
+        text = "Events",
+        fontSize = 24.sp,
+        fontWeight = FontWeight.Bold,
+        modifier = Modifier.padding(start = 16.dp, top = 6.dp, bottom = 2.dp))
+    if (eventList.events.isEmpty()) {
+      Box(modifier = Modifier.fillMaxSize()) {
+        Text(
+            text = "No events is planned yet",
```
[the remark this answers is not in the document]
Comment (SaturneV) [reply app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt]: Good catch, it's fixed!

File: app/src/main/java/com/android/streetworkapp/model/park/Park.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.android.streetworkapp.model.park
```
[the remark this answers is not in the document]
Comment (SaturneV) [reply app/src/main/java/com/android/streetworkapp/model/park/Park.kt:1]: I have refactored `model.parks` to `model.parklocation`, is it good for you?

[the remark this answers is not in the document]
Comment (alvaro1080) [reply app/src/main/java/com/android/streetworkapp/model/park/Park.kt:1]: It is perfect, thank you!

**Review by alvaro1080 — APPROVED** [review]
Thank you for the changes!

**Pull request merged**